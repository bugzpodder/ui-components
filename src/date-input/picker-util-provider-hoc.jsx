// @flow
import MomentUtils from "material-ui-pickers/utils/moment-utils";
import MuiPickersUtilsProvider from "material-ui-pickers/utils/MuiPickersUtilsProvider";
import React from "react";

export function wrapPickerUtilProvider<P>(WrappedComponent: React$ComponentType<P>): React$ComponentType<P> {
  return (props: P) => {
    return (
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}
