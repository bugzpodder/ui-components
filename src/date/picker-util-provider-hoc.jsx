// @flow
import MomentUtils from "@date-io/moment";
import React from "react";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";

export function wrapPickerUtilProvider<P>(WrappedComponent: React$ComponentType<P>) {
  return (props: P) => {
    return (
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}
