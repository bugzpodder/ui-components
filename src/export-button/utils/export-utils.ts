import {
  ExportableColumn,
  ReportOptions,
  toDelimitedReport,
} from "@grailbio/lib";
import { saveAs } from "file-saver";

export const generateReport = (
  fileName: string,
  columns: ExportableColumn<any>[],
  data: Record<string, any>[],
  options: ReportOptions = {},
): void => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
