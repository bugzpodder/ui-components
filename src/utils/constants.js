// @flow
export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  value: suggestion.label,
  label: suggestion.label,
}));
