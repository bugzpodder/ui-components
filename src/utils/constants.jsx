// @flow
import React from "react";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const SERGEY =
  "https://media.licdn.com/dms/image/C5603AQGlBIPst1V4Ew/profile-displayphoto-shrink_800_800/0?e=1535587200&v=beta&t=2V2knyEhbTrhRWzgydsXLZqiX8gnoD1poj8jGeFpA0w";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];
