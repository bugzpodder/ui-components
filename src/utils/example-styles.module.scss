.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

