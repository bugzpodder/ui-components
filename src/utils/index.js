// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
