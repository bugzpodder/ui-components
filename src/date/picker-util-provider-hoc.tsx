import MomentUtils from "@date-io/moment";
import React, { ComponentType, useContext } from "react";
import {
  DATE_FORMAT,
  DATE_TIME_FORMAT,
  DATE_TIME_UNICODE_FORMAT,
  DATE_UNICODE_FORMAT,
} from "@grailbio/lib";
import {
  MuiPickersContext,
  MuiPickersUtilsProvider,
} from "@material-ui/pickers";
import {
  MuiPickersContext as OldMuiPickersContext,
  MuiPickersUtilsProvider as OldMuiPickersUtilsProvider,
} from "material-ui-pickers";
import { ParsableDate } from "@material-ui/pickers/constants/prop-types";

export function wrapPickerUtilProvider<P>(
  WrappedComponent: ComponentType<P>,
  useOldPicker = false,
  utils = MomentUtils,
) {
  return (props: P) => {
    return useOldPicker ? (
      <OldMuiPickersUtilsProvider utils={utils}>
        <WrappedComponent {...props} />
      </OldMuiPickersUtilsProvider>
    ) : (
      <MuiPickersUtilsProvider utils={utils}>
        <WrappedComponent {...props} />
      </MuiPickersUtilsProvider>
    );
  };
}

export const useMuiPickersContext = () => {
  const utils = useContext(MuiPickersContext);
  const oldUtils = useContext(OldMuiPickersContext);

  return utils || oldUtils;
};

export const useDateFormat = () => {
  const dateUtils = useMuiPickersContext();
  if (
    dateUtils &&
    dateUtils.yearFormat &&
    DATE_UNICODE_FORMAT.indexOf(dateUtils.yearFormat) >= 0
  ) {
    return {
      defaultDateFormat: DATE_UNICODE_FORMAT,
      defaultDateTimeFormat: DATE_TIME_UNICODE_FORMAT,
    };
  }

  return {
    defaultDateFormat: DATE_FORMAT,
    defaultDateTimeFormat: DATE_TIME_FORMAT,
  };
};

export const useFormattedDateForDisplay = (
  value: ParsableDate,
  format: string,
): string => {
  const dateUtils = useMuiPickersContext();
  const date = dateUtils ? dateUtils.date(value) : value;
  if (value && dateUtils && dateUtils.isValid(date)) {
    return dateUtils.format(date, format);
  }
  return "";
};
