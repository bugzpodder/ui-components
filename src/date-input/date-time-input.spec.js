// @flow
import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { TestWrapper } from "../utils";
import { wrapPickerUtilProvider } from "./picker-util-provider-hoc";
import { DateTimeInput } from "./index";

const DateTimeInputContainer = wrapPickerUtilProvider(DateTimeInput);

test("render date time input", () => {
	const testDate = "2016-03-07 16:20:00";
	const mockCallback = jest.fn();
	const { container, getByPlaceholderText } = render(
		<TestWrapper>
			<DateTimeInputContainer
				placeholder="Test DateTime Input"
				onChange={mockCallback}
				value={testDate} />
		</TestWrapper>,
	);
	expect(getByPlaceholderText("Test DateTime Input")).toHaveAttribute("value", "2016-03-07 16:20:00");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true", async () => {
	const testDate = "2017-03-07 16:20:00";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateTimeInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly={true} />
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("2017-03-07 16:20:00");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true showing - for empty value", async () => {
	const testDate = "";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateTimeInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly />
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("-");
	expect(container).toMatchSnapshot();
});

test("render date input when readOnly is true showing empty value", async () => {
	const testDate = "";
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<DateTimeInputContainer
				placeholder="Test Date Input"
				onChange={mockCallback}
				value={testDate}
				readOnly
				showEmptyValue
			/>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("");
	expect(container).toMatchSnapshot();
});
