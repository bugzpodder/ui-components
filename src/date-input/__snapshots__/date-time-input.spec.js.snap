// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render date input when readOnly is true 1`] = `
<div>
   
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    2017-03-07 16:20:00
  </span>
   
</div>
`;

exports[`render date input when readOnly is true showing - for empty value 1`] = `
<div>
   
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    -
  </span>
   
</div>
`;

exports[`render date input when readOnly is true showing empty value 1`] = `
<div>
   
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    
  </span>
   
</div>
`;

exports[`render date time input 1`] = `
<div>
   
  <div
    class="MuiFormControl-root"
  >
    <div
      class="MuiInput-root MuiInput-formControl MuiInput-underline WithUtils-DateTextField--input-2"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input"
        placeholder="Test DateTime Input"
        type="text"
        value="2016-03-07 16:20:00"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
               
              event
               
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    
  </div>
   
</div>
`;

exports[`render null date time input 1`] = `
<div>
   
  <div
    class="MuiFormControl-root"
  >
    <div
      class="MuiInput-root MuiInput-formControl MuiInput-underline WithUtils-DateTextField--input-66"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input"
        placeholder="Test DateTime Input"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
               
              event
               
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    
  </div>
   
</div>
`;
